roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -437,54 +451,94 @@ impl JobDeclarator {
         new_block_sender: Sender<String>,
         sender_add_txs_to_mempool: Sender<AddTrasactionsToMempoolInner>,
     ) {
-        let listner = TcpListener::bind(&config.listen_jd_address).await.unwrap();
-        while let Ok((stream, _)) = listner.accept().await {
+        let listener = TcpListener::bind(&config.listen_jd_address).await.unwrap();
+
+        while let Ok((stream, _)) = listener.accept().await {
             let responder = Responder::from_authority_kp(
                 &config.authority_public_key.into_bytes(),
                 &config.authority_secret_key.into_bytes(),
                 std::time::Duration::from_secs(config.cert_validity_sec),
             )
             .unwrap();
+
             let addr = stream.peer_addr();
 
             if let Ok((receiver, sender, _, _)) =
                 Connection::new(stream, HandshakeRole::Responder(responder)).await
             {
-                let setup_message_from_proxy_jd = receiver.recv().await.unwrap();
-                info!(
-                    "Setup connection message from proxy: {:?}",
-                    setup_message_from_proxy_jd
-                );
+                match receiver.recv().await {
+                    Ok(EitherFrame::Sv2(mut sv2_message)) => {
+                        debug!("Received SV2 message: {:?}", sv2_message);
+                        let payload = sv2_message.payload();
 
-                let setup_connection_success_to_proxy = SetupConnectionSuccess {
-                    used_version: 2,
-                    // Setup flags for async_mining_allowed
-                    flags: 0b_0000_0000_0000_0000_0000_0000_0000_0001,
-                };
-                let sv2_frame: StdFrame =
-                    JdsMessages::Common(setup_connection_success_to_proxy.into())
-                        .try_into()
-                        .unwrap();
-                let sv2_frame = sv2_frame.into();
-                info!("Sending success message for proxy");
-                sender.send(sv2_frame).await.unwrap();
+                        if let Ok(setup_connection) =
+                            binary_sv2::from_bytes::<SetupConnection>(payload)
+                        {
+                            let flag = setup_connection.flags;
+                            let is_valid = SetupConnection::check_flags(
+                                Protocol::JobDeclarationProtocol,
+                                flag,
+                                1,
+                            );
```

Comment:
it seems we were mixing up the order of the parameters:

second parameter is `available_flags`, which signals that JDS supports async jobs, and therefore should be `1`

third parameter is `required_flags`,  which signals if JDC requested async jobs or not, so that one should be `flag`
```suggestion
                            let is_valid = SetupConnection::check_flags(
                                Protocol::JobDeclarationProtocol,
                                flag,
                                1,
                            );
```